ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -147,7 +147,21 @@ function rhel_upgrade(){
 
 # Extract oc binary which is supposed to be identical with target release
 function extract_oc(){
-    local url; url="openshift-release-artifacts.apps.ci.l2s4.p1.openshiftapps.com"
+    local route="openshift-release-artifacts"
+    local url
+    local target_arch=""
+    if [ "${OCP_ARCH}" != "amd64" ]; then
+      # Change the route to use for non-amd64 clusters
+      route="${route}-${OCP_ARCH}"
+      if [ "$(uname -m)" = "x86_64" ]; then
```

Comment:
I'm trying to understand the code block. We have both [openshift-client-linux-4.12.0-0.nightly-arm64-2022-10-20-100222.tar.gz](https://openshift-release-artifacts-arm64.apps.ci.l2s4.p1.openshiftapps.com/4.12.0-0.nightly-arm64-2022-10-20-100222/openshift-client-linux-4.12.0-0.nightly-arm64-2022-10-20-100222.tar.gz) and [openshift-client-linux-amd64-4.12.0-0.nightly-arm64-2022-10-20-100222.tar.gz](https://openshift-release-artifacts-arm64.apps.ci.l2s4.p1.openshiftapps.com/4.12.0-0.nightly-arm64-2022-10-20-100222/openshift-client-linux-amd64-4.12.0-0.nightly-arm64-2022-10-20-100222.tar.gz). What's the difference between them? Why we need amd64 client in an arm cluster?

Reply:
Some binaries are not compatible. E.g, cp, mv
So actually we are running the installation in a amd64 pod/cluster to set up the ARM cluster.

Reply:
Currently we run oc in amd64 pods against arm64 clusters. However, we are also working at having another prow cluster in the build farm, arm01, to enable testing from arm pods. This code is future proof to allow running some of the jobs in the arm01 cluster when it will be ready.

Reply:
ack. thank you both!